pkg/modprovider/module.go

```diff
@@ -0,0 +1,482 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package modprovider
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/pulumi/pulumi/pkg/v3/resource/provider"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/urn"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+
+	"github.com/pulumi/pulumi-terraform-module/pkg/auxprovider"
+	"github.com/pulumi/pulumi-terraform-module/pkg/tfsandbox"
+)
+
+var useCustomResource bool = cmdutil.IsTruthy(os.Getenv("PULUMI_TERRAFORM_MODULE_CUSTOM_RESOURCE"))
```

Comment:
Tempted to call it PULUMI_VIEWS

Reply:
Or align with what we're using in the CLI (`PULUMI_ENABLE_VIEWS_PREVIEW`) since that'll have to be set too https://github.com/pulumi/pulumi/blob/fe175b6b035c3ed053800485bf1bc6daac363ddb/sdk/go/common/resource/plugin/provider_plugin.go#L55-L56

Reply:
(That too is overly verbose... we could shorten it, but it kinda doesn't matter because it isn't long for this world)